From 2245a0d18500d3f0c8a857f8a484e20e6913104b Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sat, 12 Dec 2015 21:59:51 -0800
Subject: [PATCH 061/192] SnapdragonCamera: Set camera parameters before
 restarting preview

Ensure that all of the updated camera parameters are propagated to the camera
HAL prior to restarting the preview. This fixes the split-exposure observed when
turning ZSL off and on without exiting the app.

Change-Id: I392549ae7cc75b150e01cf8b635d2e3f1ad6ffb9
Signed-off-by: Sultanxda <sultanxda@gmail.com>
---
 src/com/android/camera/PhotoModule.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index b5a8ac3c3..4bab7db84 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -4952,6 +4952,7 @@ public class PhotoModule
         mLocationManager.recordLocation(recordLocation);
         if(needRestart()){
             Log.v(TAG, "Restarting Preview... Camera Mode Changed");
+            setCameraParameters(UPDATE_PARAM_PREFERENCE);
             stopPreview();
             startPreview();
             setCameraState(IDLE);
-- 
2.17.1

